design-proposals/eim-cli.md

```diff
@@ -0,0 +1,254 @@
+
+### Ability to register/onboard/provision a host
+
+#### Local Options
+
+The local options associated with the `host` object/noun would help with filtering and providing arguments to the commands:
+
+- `-o` `registered`/`onboarded`/`provisioned`/`all`/`deauthorized`/`unknown`/`error` - helps filter out the output of `list` command/verb ( TODO maybe instead of `-o` for output change to `-s` for status???)
+- `--import-from-csv` `my-hosts.csv` - used to provision a bulk of hosts using BIT
+> **NOTE**: TODO - Team I am not yet convinced on using the register/onboard/provision flags as options to the create/delete objects/nouns rather than using as verbs directly - will welcome feedback
+- `--register` `<arguments>` - local flag for the `create` command - enables only registering host during create - takes in argument in particular order necessary for registering the host 
+- `--onboard` - local flag for the `create` command - enables only onboarding of already registered host during create
+- `--provision` `<arguments>` - local flag for the `create` command - enables only provisioning of already onboarded host during create - takes in argument in particular order necessary for provisioning of the host
+- `--auto-provision` - optional local flag that only takes effect when `--register` flag is invoked - enables auto provisioning of node when it connects
+> **Note**: TODO - If it is the case that the autoprovision toggle in registration view in UI is ignored if the provisioner is set to autoprovision (TODO also get to to bottom why the toggle is allowed in UI if that is the case) then the CLI should spit out a warning message about this whenever a host is being created/registered/onboarded
+- `--auto-onboarding` - optional local flag that only takes effect when `--register` flag is invoked - enables auto onboarding of node when it connects
+- `--deauthorize-only` - optional local flag that only takes effect when `delete` flag is invoked - does not delete host fully but only deauthorizes it
+- `--force-delete` - optional local flag that only takes effect when `delete` flag is invoked - deletes host without deauthorization
+- `--edit-config` `<arguments>` - set of arguments to aid the edition of host
+
+#### List  host information
+
+The list command is responsible for listing all deployed hosts, in tabular format with the amount of information equivalent to what is present in UI: `Name`/`Host Status`/`Serial Number`/`Operating System`/`Site`/`Workload`/`Actions`/`UUID`/`Processor`/`Latest Updates`/`Trusted Compute`   
+> **TODO** this list should probably be scaled back by default and allow for expansion with `-o wide`? Looking for feedback.
+
+- List Registered Host (ie. `emfctl list hosts -o registered`)
+- List Onboarded Host (ie. `emfctl list hosts -o onboarded`)
+- List Provisioned Host (ie. `emfctl list hosts -o provisioned`)
+- List All Host (ie. `emfctl list hosts`) (the default behaviour)
```

Comment:
we should consider also the chance of complex filter logic (dirrectly exposing our filter engine) - maybe is loaded from a file to avoid issues caused by quotes

Reply:
also in terms of pre-baked filter we should look at all the lists commands issued by the UI + summary call that can be a shorten version of this command